src/main/java/com/google/firebase/remoteconfig/FirebaseRemoteConfig.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.firebase.remoteconfig;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.core.ApiFuture;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.ImplFirebaseTrampolines;
+import com.google.firebase.internal.CallableOperation;
+import com.google.firebase.internal.FirebaseService;
+
+/**
+ * This class is the entry point for all server-side Firebase Remote Config actions.
+ *
+ * <p>You can get an instance of FirebaseRemoteConfig via {@link #getInstance(FirebaseApp)}, and
+ * then use it to manage Remote Config templates.
+ */
+public final class FirebaseRemoteConfig {
+
+  private static final String SERVICE_ID = FirebaseRemoteConfig.class.getName();
+  private final FirebaseApp app;
+  private final Supplier<? extends FirebaseRemoteConfigClient> remoteConfigClient;
+
+  private FirebaseRemoteConfig(Builder builder) {
```

Comment:
A builder is overkill for just 2 arguments. I'd recommend something simpler and direct:

```java
@VisibleForTesting
FirebaseRemoteConfig(FirebaseApp app, FirebaseRemoteConfigClient client) {
  this.app = checkNotNull(app);
  this.client = checkNotNull(client);
}

private FirebaseRemoteConfig(FirebaseApp app) {
  this(app, FirebaseRemoteConfigClientImpl.fromApp(app));
}
```

Note that I'm also not using a Supplier here. Since the client is needed for all operations in this API, there's no reason to lazy initialize it.

Reply:
Good point! Updated the code.
I had to update the unit test for `testRemoteConfigClientWithoutProjectId()`. As the client is now initialized at construction it throws at `FirebaseRemoteConfig.getInstance()`.